host/tests/deepsleepClasses/power.py

```diff
@@ -0,0 +1,233 @@
+
+        # Convert input arguments to strings and update the prompts
+        promptWithAnswers[0]["input"] = str(mode)
+        result = self.utMenu.select( self.testSuite, "Set Power State", promptWithAnswers)
+
+    def getPowerMode(self):
+        """
+        Get Power Mode.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select( self.testSuite, "Get Power State")
+
+    def setWakeupSource(self, source:int, enable:int):
+        """
+        Set wakeup source.
+
+        Args:
+            source: int: This enumeration defines the wakeup source.
+                            Refer to the PWRMGR_WakeupSrcType_t enum for valid options
+            enable: int: Enable/Disable Wakeup Source
+
+        Returns:
+            None
+        """
+
+        promptWithAnswers = [
+            {
+                "query_type": "direct",
+                "query": "Select Wake Up Source Type: ",
+                "input": "PWRMGR_WAKEUPSRC_VOICE"
+            },
+            {
+                "query_type": "direct",
+                "query": "Enable/Disable Wakeup Source",
+                "input": "1"
+            }
+        ]
+
+        # Convert input arguments to strings and update the prompts
+        promptWithAnswers[0]["input"] = str(source)
+        promptWithAnswers[1]["input"] = str(enable)
+        result = self.utMenu.select( self.testSuite, "Set Wakeup Source", promptWithAnswers)
+
+    def getWakeupSource(self, source:int):
+        """
+        Get wakeup source.
+
+        Args:
+            source: int: This enumeration defines the wakeup source.
+                            Refer to the PWRMGR_WakeupSrcType_t enum for valid options
+
+        Returns:
+            None
+        """
+
+        promptWithAnswers = [
+            {
+                "query_type": "direct",
+                "query": "Select Wake Up Source Type: ",
+                "input": "PWRMGR_WAKEUPSRC_VOICE"
+            }
+        ]
+
+        # Convert input arguments to strings and update the prompts
+        promptWithAnswers[0]["input"] = str(source)
+        result = self.utMenu.select( self.testSuite, "Get Wakeup Source", promptWithAnswers)
+
+    def terminate(self):
+        """
+        Terminates the Power manager module.
+
+        Args:
+            None.
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select(self.testSuite, "Terminate Power Manager")
+
+    def __del__(self):
+        """
+        De-Initializes the power manager helper function.
+
+        Args:
+            None.
+
+        Returns:
+            None
+        """
+        self.utMenu.stop()
+
+# Test and example usage code
+if __name__ == '__main__':
+
+    shell = InteractiveShell()
+    shell.open()
+
+    platformProfile = "https://github.com/rdkcentral/rdk-halif-test-power_manager/blob/main/profiles/source/source_powerManager.yaml" #Does this work? Do we need something else?
```

Comment:
http wont work
yaml file is part of the test, you can give relative path to the profile file